### PR TITLE
Small updates to adcc integration

### PR DESCRIPTION
**Pull request by mfherbst**
## Description
Add rotational strength as new property, which has become available in adcc 0.14.3.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [X] Rotational strength for all ADC methods from adcc
- [x] Velocity gauge for oscillator strength

## Questions
- [x] Adcc now allows oscillator strengths to be computed in both velocity and length gauge. What is the canonical way to expose this in Psi4?

## Checklist
- [x] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by loriab** [comment]
Is https://github.com/psi4/psi4/blob/master/psi4/src/read_options.cc#L1998 what you're looking for?

File: psi4/driver/procrouting/proc.py [hunk 1]
```diff
@@ -3464,17 +3465,26 @@ def format_vector(label, data):
...
         adc_wfn.set_variable(name + " transition dipoles", core.Matrix.from_array(data))
 
     if "OSCILLATOR_STRENGTH" in properties:
-        data = state.oscillator_strengths.reshape(-1, 1)
-        computed["Oscillator strength (length gauge)"] = data
+        if gauge == "velocity":
+            data = state.oscillator_strengths_velocity.reshape(-1, 1)
+        else:
+            data = state.oscillator_strengths.reshape(-1, 1)
+        computed["Oscillator strength (" + gauge + " gauge)"] = data
```
Comment (robertodr) [inline psi4/driver/procrouting/proc.py:3480]: Could you use `f`-strings for these string operations?

Comment (mfherbst) [reply psi4/driver/procrouting/proc.py:3480]: With pleasure. I thought to leave it out for compatibility to 3.5.

File: psi4/driver/procrouting/proc.py [hunk 2]
```diff
@@ -3464,21 +3465,30 @@ def format_vector(label, data):
...
     if "OSCILLATOR_STRENGTH" in properties:
-        data = state.oscillator_strengths.reshape(-1, 1)
-        computed["Oscillator strength (length gauge)"] = data
-        adc_wfn.set_variable(name + " oscillator strengths", core.Matrix.from_array(data))
+        if gauge == "velocity":
+            data = state.oscillator_strengths_velocity.reshape(-1, 1)
+        else:
+            data = state.oscillator_strengths.reshape(-1, 1)
+        computed[f"Oscillator strength ({gauge} gauge)"] = data
+        adc_wfn.set_variable(f"{name} oscillator strengths", core.Matrix.from_array(data))
```
Comment (loriab) [inline psi4/driver/procrouting/proc.py:3481]: can the variable include the gauge like https://github.com/psi4/psi4/blob/master/psi4/src/psi4/cc/ccresponse/optrot.cc#L442-L514

File: tests/adcc/CMakeLists.txt [hunk 3]
```diff
@@ -18,3 +18,6 @@ add_subdirectory(h2o-adc2-any)
 add_subdirectory(cn-adc1)
 add_subdirectory(cn-adc2)
 add_subdirectory(hf-adc2-spin-flip)
+
+# Specific properties
+add_subdirectory(methyloxirane_rotation)
```
Comment (loriab) [inline tests/adcc/CMakeLists.txt:23]: I think you're mixing underscore and dash in the dir/test name. go for dash, please.

File: psi4/driver/procrouting/proc.py [hunk 4]
```diff
@@ -3464,21 +3465,38 @@ def format_vector(label, data):
...
+        else:
+            data = state.oscillator_strengths.reshape(-1, 1)
+        computed[f"Oscillator strength ({gauge} gauge)"] = data
+        adc_wfn.set_variable(f"{name} oscillator strengths ({gauge_short})",
+                             core.Matrix.from_array(data))
+
+    if "ROTATIONAL_STRENGTH" in properties:
+        data = state.rotatory_strengths.reshape(-1, 1)
+        computed["Rotational strength"] = data
+        adc_wfn.set_variable(f"{name} rotational strengths", core.Matrix.from_array(data))
```
Comment (hokru) [inline psi4/driver/procrouting/proc.py:3494]: Could you also add the gauge to the rotational strength variable? To have consistency. Whatever is used in adcc.

**Comment by mfherbst** [comment]
@hokru  Just let me know when a proper rebase can be done.

**Pull request merged**